pkg/controller/controller.go

```diff
@@ -512,6 +514,42 @@ func unmarshalParameters(in []byte) (map[string]interface{}, error) {
 	return parameters, nil
 }
 
+// fetchSecretParameters requests and returns the contents of the given secret as a map
+func fetchSecretParameters(kubeClient kubernetes.Interface, namespace string, secretRef *v1.LocalObjectReference) (map[string]interface{}, error) {
+	if kubeClient == nil {
+		return nil, errors.New("kubeClient is not defined")
+	}
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]interface{})
+	for k, v := range secret.Data {
+		var parameters interface{}
+		if json.Unmarshal(v, &parameters) != nil {
```

Comment:
@duglin no, it's raw byte array there. If I recall correctly, the transformation into/from base64 is being done by `kubectl` on-the-fly (when you ask for `-o yaml`), and you won't see it with raw REST API.

See binging credentials code, for example - it doesn't use base64 encode either: https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/controller/controller_binding.go#L402